### PR TITLE
Display version on client

### PR DESCRIPTION
**Pull request by ZachMyers3**
For #216.

Just a simple API endpoint from the server. Currently pulling and displaying on the login prompt's legend. I figured there were other places better suited for this but wanted input before throwing them around.

Let me know what you think!

### CODE REVIEW

**Comment by ZachMyers3** [comment]
Here's a screenshot of the display.

image

File: server/api/http/version.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from aiohttp import web
+
+async def get_version(request: web.Request):
+    try:
+        with open('VERSION', 'r') as version_file:
+            version_data = version_file.read()
+    except:
+        version_data = "0.0.0"
```
Comment (Kruptein) [inline server/api/http/version.py]: I would just return a 500 error because if this happens, there is something wrong with the server.

**Comment by ZachMyers3** [comment]
If we want to inject it into the initialization state, should we tie it into the authentication query? Otherwise we'd await two responses on initialization.

**Comment by Kruptein** [comment]
I get your point, but I think that from an API standpoint, returning version info in the is_authed endpoint is strange.

you can use Promise.all to start two parallel request that won't block on eachother.

**Comment by ZachMyers3** [comment]
I definitely agree with you, I think this is the cleaner way to do it. 

Latest commit has it this way.

File: client/src/router.ts [hunk 2]
```diff
@@ -75,14 +75,17 @@ export const router = new Router({
 router.beforeEach(async (to, _from, next) => {
     coreStore.setLoading(true);
     if (!coreStore.initialized) {
-        const response = await fetch("/api/auth");
-        if (response.ok) {
-            const data = await response.json();
-            if (data.auth) {
+        const authResponse = await fetch("/api/auth");
+        const versionResponse = await fetch("/api/version");
+        if (authResponse.ok && versionResponse.ok) {
```
Comment (Kruptein) [inline client/src/router.ts]: Now you block the second request by waiting on the first.  What you want to do is something akin to

```
const [authResponse, versionResponse] = await Promise.all([fetch("/api/auth"), fetch("/api/version"]);
coreStore.setVersion(versionResponse.version);
if (authResponse.auth) { ... }
```

File: server/api/http/version.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from aiohttp import web
+
+async def get_version(request: web.Request):
+    try:
+        with open('VERSION', 'r') as version_file:
+            version_data = version_file.read()
+    except:
+        return web.HTTPError()
```
Comment (Kruptein) [inline server/api/http/version.py]:
```suggestion
        return web.HTTPInternalServerError(reason="Version file could not be loaded")
```

File: client/src/router.ts (hunk 2 above)
Comment (ZachMyers3) [reply client/src/router.ts]: Latest commit has these changes.

I've actually never done this before, so it's new!

**Comment by ZachMyers3** [comment]
Works for me! Thanks!

**Comment by Kruptein** [comment]
I did just notice that somehow, the package-lock is empty in dev.  Could you reinstantiate that file? :D

**Comment by Kruptein** [comment]
On second thoughts, I'll just update it on dev immediately

**Pull request merged**